src/metabase/models/interface.clj

```diff
@@ -251,6 +252,22 @@
   {:in  json-in
    :out json-out-without-keywordization})
 
+(defn assert-enum
+  "Assert that a value is one of the values in `enum`."
+  [enum value]
+  (assert (set? enum) "enum but be a set")
```

Comment:
I don't think we have asserts disabled in production? If not, the overhead is just something to bear in mind. It's arguably a bit nicer to just use `mu/defn` and type it directly.